app/models/manageiq/providers/azure/cloud_manager/provision/cloning.rb

```diff
@@ -10,8 +10,11 @@ def do_clone_task_check(clone_task_ref)
   end
 
   def find_destination_in_vmdb(vm_uid_hash)
-    ems_ref = vm_uid_hash.values.join("\\")
-    ManageIQ::Providers::Azure::CloudManager::Vm.find_by(:ems_ref => ems_ref.downcase)
+    # Azure vm name is case-case-sensitive, we don't want to downcase it
+    vm_name = vm_uid_hash.delete(:vm_name)
+    ref = vm_uid_hash.values.join('\\').downcase
+    ref << '\\' + vm_name
```

Comment:
for record, thoses faild test are here 
https://travis-ci.org/ManageIQ/manageiq/jobs/252011632